test/README.md

```diff
@@ -0,0 +1,55 @@
+# Single Asset Staking Unit Tests Documentation
+
+This offers a comprehensive guide to the Single Asset Staking unit tests within the project. The tests are meticulously designed to assess the accuracy and efficiency of the Single Asset Staking implementation, particularly focusing on the validation of actions involved during the staking  process.
+
+## Overview
+
+The unit tests for the Single Asset Staking are organized into a suite titled "Staking Unit Test." This suite is dedicated to verifying the validation process for the various actions involved in orchestrating and participating in staking.
+
+## Test Suite Details
+
+### Tests Included
+
+-**Pass - Init Staking**: This test confirms the proper initialization of Staking by creating of a head node for the linked list which will contain all the staking requests and stake  amount.
+-**Pass - Deinit Staking**: This confirms correct working validation of deinitization of head node once the staking rewards have been processed correctly.
+-**Pass - Insert Staking**: Aims to validate a new user's request to participate in Staking, confirming its correct positioning and validation.
+-**Pass - Remove Staking**: Aims to validate an existing user's request to discontinue participation (before stake is frozen) in Staking and reclaim their stake. It also checks that the removal happens without compromising the integrity of the linked list.
+-**Pass - Late Remove Staking**: Aims to validate an existing user's request to discontinue participation in Staking (after stake is frozen and before staking ends) and reclaim their stake. It checks that a penalty fee is paid for withdrawing late from Staking but is still allowed. Additionally ensuring that removal happens without compromising the integrity of the linked list.
+-**Pass - Claim Stake & Reward**: This test confirms that a user is able to successfully claim their stake and respective reward after the end of the staking period.
+
+## Running the Tests
+
+To execute the Staking unit tests for the project, you should follow the standard testing procedures outlined in the project documentation. Typically, this involves executing a command such as:
+
+```sh
+cabal new-test --test-show-details=streaming
+```
+
+This command will compile and execute all the test suites defined in the project. The output will show the status of each test case.
+
+### Test Outcome Summary
+
+In the most recent execution:
```

Comment:
We might want to consider adding a link here to the project's github actions page.